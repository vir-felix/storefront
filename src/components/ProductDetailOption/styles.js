export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
