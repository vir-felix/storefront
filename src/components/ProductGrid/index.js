export { default as ProductGrid } from "./ProductGrid";
export { default as ProductItem } from "./ProductItem";
