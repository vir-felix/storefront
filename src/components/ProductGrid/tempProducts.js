export const tempProducts = [
  {
    _id: "ioKwBYk6Nom9K92cv",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Scout Book",
    originCountry: "US",
    requiresShipping: true,
    handle: "reaction-scout-book",
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "0.00",
      min: 0,
      max: 0
    },
    workflow: {
      status: "new"
    },
    isLowQuantity: true,
    isSoldOut: false,
    vendor: "Bookworks",
    description:
      "Need to make a quick note? Got a once-in-a-lifetime idea? Write it down in one of these great little notebooks",
    isBackorder: false,
    hashtags: ["hZdSevCBP8dDc66Ba"],
    createdAt: "2018-04-12T18:48:28.969Z",
    updatedAt: "2018-04-12T18:52:11.083Z",
    positions: {
      reaction: {
        weight: 2,
        updatedAt: "2018-04-12T18:52:08.844Z",
        position: 0
      }
    },
    weight: 2,
    media: []
  },
  {
    _id: "BCTMZ6HTxFSppJESk",
    title: "Basic Reaction Product",
    shopId: "J8Bhq3uTtdgwZx3rz",
    ancestors: [],
    description:
      "Sign in as administrator to edit.\nYou can clone this product from the product grid.\nYou can upload images click or drag in image box on the left here.\nTag this product below, and then add tag in navigation.\nClick the bookmark in the tag to set product url.\nOption variants, price, quantity, and child variants are created by clicking on the variant below, clone the variant to add more options.\nDetails can be added below the image for more specific product information.\n Login next to the cart, and then click the dashboard icon for more tools.",
    hashtags: ["rpjCvTBGjhBi2xdro", "cseCBSSrJ3t8HQSNP"],
    price: {
      range: "12.99 - 19.99",
      min: 12.99,
      max: 19.99
    },
    isVisible: true,
    isLowQuantity: false,
    isSoldOut: true,
    isBackorder: true,
    metafields: [
      {
        key: "Material",
        value: "Cotton"
      },
      {
        key: "Quality",
        value: "Excellent"
      }
    ],
    pageTitle: "This is a basic product. You can do a lot with it.",
    type: "product-simple",
    updatedAt: "2018-04-12T18:52:17.881Z",
    vendor: "Example Manufacturer",
    originCountry: "US",
    requiresShipping: true,
    handle: "example-product",
    isDeleted: false,
    template: "productDetailSimple",
    createdAt: "2018-04-12T17:53:15.134Z",
    workflow: {
      status: "new"
    },
    weight: 0,
    media: [],
    positions: {
      reaction: {
        weight: 1,
        updatedAt: "2018-04-12T18:52:08.855Z",
        position: 1
      }
    }
  },
  {
    _id: "h2v7MYCXBtjMm7Piv",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Bag of Beans",
    originCountry: "US",
    requiresShipping: true,
    handle: "reaction-bag-of-beans",
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "12 - 24",
      min: 12,
      max: 24
    },
    workflow: {
      status: "new"
    },
    isLowQuantity: false,
    isSoldOut: false,
    vendor: "Bluestone Coffee",
    description:
      "A bag of rich Arabica coffee. Great for a fueling up for a night out or for settling in for a night of coding at home",
    isBackorder: false,
    positions: {
      "reaction swag shop": {
        weight: 1
      },
      reaction: {
        position: 2,
        updatedAt: "2018-04-12T18:52:08.870Z"
      }
    },
    hashtags: ["hZdSevCBP8dDc66Ba"],
    createdAt: "2018-04-12T18:48:28.870Z",
    updatedAt: "2018-04-12T18:52:20.366Z",
    weight: 1,
    media: []
  },
  {
    _id: "z6paQqnrGnWBvepyB",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Mug",
    originCountry: "US",
    requiresShipping: true,
    handle: "reaction-mug",
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "15",
      min: 15,
      max: 15
    },
    workflow: {
      status: "new"
    },
    isLowQuantity: false,
    isSoldOut: false,
    vendor: "Mugworks",
    description: "A sturdy mug to keep your hot drinks hot and your cold drinks cold",
    isBackorder: false,
    positions: {
      "reaction swag shop": {
        weight: 1
      },
      reaction: {
        position: 3,
        updatedAt: "2018-04-12T18:52:08.879Z"
      }
    },
    hashtags: ["hZdSevCBP8dDc66Ba"],
    createdAt: "2018-04-12T18:48:28.936Z",
    updatedAt: "2018-04-12T18:52:22.147Z",
    weight: 0,
    media: []
  },
  {
    _id: "gygxrADdPPWoCCmS7",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Commerce Pouch",
    originCountry: "US",
    requiresShipping: true,
    handle: "reaction-commerce-pouch",
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "8",
      min: 8,
      max: 8
    },
    workflow: {
      status: "new"
    },
    vendor: "Pouchworks",
    description: "Need a small pouch to keep things in? This is your ticket",
    isLowQuantity: false,
    isSoldOut: true,
    isBackorder: false,
    hashtags: ["hZdSevCBP8dDc66Ba"],
    createdAt: "2018-04-12T18:48:28.954Z",
    updatedAt: "2018-04-12T18:52:24.886Z",
    positions: {
      reaction: {
        weight: 1,
        updatedAt: "2018-04-12T18:52:08.910Z",
        position: 4
      }
    },
    weight: 0,
    media: []
  },
  {
    _id: "xsudQc3K2EbqCeFBm",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Ladies Shirt",
    originCountry: "US",
    requiresShipping: true,
    handle: "reaction-ladies-shirt",
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "18 - 20",
      min: 18,
      max: 20
    },
    workflow: {
      status: "new"
    },
    isLowQuantity: false,
    isSoldOut: false,
    vendor: "Shirtworks",
    description:
      "The same hip Reaction logo in a cut more suitable for the ladies. 100% cotton fabric and double-stitching make this a shirt that will last for years to come",
    isBackorder: false,
    hashtags: ["dhqmJjXWotCNkTzxg"],
    createdAt: "2018-04-12T18:48:28.984Z",
    updatedAt: "2018-04-12T18:52:28.039Z",
    weight: 0,
    media: [],
    positions: {
      reaction: {
        position: 5,
        updatedAt: "2018-04-12T18:52:08.916Z"
      }
    }
  },
  {
    _id: "GpZM5v5qQBCdXsoqW",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Mens Shirt",
    originCountry: "US",
    requiresShipping: true,
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "18 - 20",
      min: 18,
      max: 20
    },
    workflow: {
      status: "new"
    },
    isLowQuantity: false,
    isSoldOut: false,
    vendor: "Shirtworks",
    description:
      "Get your Reaction together and show it to the world. 100% cotton fabric and double-stitching make this a shirt that will last for years to come",
    isBackorder: false,
    handle: "reaction-mens-shirt",
    hashtags: ["dhqmJjXWotCNkTzxg"],
    createdAt: "2018-04-12T18:48:29.012Z",
    updatedAt: "2018-04-12T18:52:31.154Z",
    weight: 0,
    media: [],
    positions: {
      reaction: {
        position: 6,
        updatedAt: "2018-04-12T18:52:08.948Z"
      }
    }
  },
  {
    _id: "9zxhE27pAvx8vbKr5",
    type: "product-simple",
    ancestors: [],
    shopId: "J8Bhq3uTtdgwZx3rz",
    title: "Reaction Sticker",
    originCountry: "US",
    requiresShipping: true,
    handle: "reaction-sticker",
    isDeleted: false,
    isVisible: true,
    template: "productDetailSimple",
    price: {
      range: "4",
      min: 4,
      max: 4
    },
    workflow: {
      status: "new"
    },
    isLowQuantity: false,
    isSoldOut: false,
    vendor: "Stickerworks",
    description: "Show your pride and affix some Reaction to almost any surface",
    isBackorder: false,
    hashtags: ["hZdSevCBP8dDc66Ba"],
    createdAt: "2018-04-12T18:48:29.039Z",
    updatedAt: "2018-04-12T18:52:34.783Z",
    positions: {
      reaction: {
        weight: 2,
        updatedAt: "2018-04-12T18:52:08.986Z",
        position: 7
      }
    },
    weight: 1,
    media: []
  }
];
