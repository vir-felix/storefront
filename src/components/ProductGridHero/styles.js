export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
