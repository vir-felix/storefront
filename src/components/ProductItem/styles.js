export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
