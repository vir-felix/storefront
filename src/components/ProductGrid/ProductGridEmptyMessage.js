import React, { Component } from "react";
import PropTypes from "prop-types";
import { inject } from "mobx-react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Link from "components/Link";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    marginTop: "4rem"
  },
  actionMessage: {
    color: theme.palette.reaction.coolGrey400
  },
  notFoundMessage: {
    color: theme.palette.reaction.black65
  }
});

@withStyles(styles)
@inject("routingStore")
export default class ProductGridEmptyMessage extends Component {
  static propTypes = {
    actionMessage: PropTypes.string,
    classes: PropTypes.object,
    notFoundMessage: PropTypes.string,
    routingStore: PropTypes.object
  }

  static defaultProps = {
    actionMessage: "Clear filters",
    notFoundMessage: "Sorry! We couldn't find what you're looking for."
  }

  render() {
    const { classes, actionMessage, notFoundMessage, routingStore } = this.props;

    let resetLink = routingStore.pathname;
    if (routingStore && routingStore.query && routingStore.query.slug) {
      resetLink = `${routingStore.pathname}/${routingStore.query.slug}`;
    }

    return (
      <div className={classes.root}>
        <Typography className={classes.notFoundMessage} paragraph>{notFoundMessage}</Typography>
        <Typography className={classes.actionMessage}>
          <Link route={resetLink}>{actionMessage}</Link>
        </Typography>
      </div>
    );
  }
}
