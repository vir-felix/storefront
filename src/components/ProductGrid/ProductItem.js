import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Chip from "material-ui/Chip";
import Typography from "material-ui/Typography";

const styles = (theme) => ({
  root: {},
  productInfo: {
    display: "flex",
    justifyContent: "space-between"
  },
  productMedia: {
    position: "relative"
  },
  chip: {
    borderRadius: 4,
    fontSize: 12,
    height: "auto",
    padding: theme.spacing.unit * 0.5,
    position: "absolute"
  },
  chipLabel: {
    fontWeight: theme.typography.fontWeightBold,
    padding: 0
  },
  status: {
    color: theme.palette.primary.contrastText,
    left: theme.spacing.unit,
    top: theme.spacing.unit
  },
  statusSoldOut: {
    backgroundColor: theme.palette.secondary.main
  },
  statusBackorder: {
    backgroundColor: theme.palette.secondary.dark
  },
  statusSale: {
    backgroundColor: theme.palette.error.main
  },
  statusBestseller: {
    backgroundColor: theme.palette.primary.light
  },
  warning: {
    backgroundColor: "transparent",
    ...theme.typography.caption,
    right: theme.spacing.unit,
    top: theme.spacing.unit
  },
  img: {
    height: "auto",
    width: "100%"
  }
});

@withStyles(styles)
class ProductItem extends Component {
  static propTypes = {
    classes: PropTypes.object,
    product: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    product: {}
  };

  get productStatus() {
    const { classes, product: { isBackorder, isSoldOut } } = this.props;
    let status;
    if (isSoldOut && isBackorder) {
      status = { label: "Backorder", style: `${classes.status} ${classes.statusBackorder}` };
    } else if (isSoldOut && !isBackorder) {
      status = { label: "Sold Out", style: `${classes.status} ${classes.statusSoldOut}` };
    }
    return status;
  }

  get productLowQuantity() {
    const { product: { isLowQuantity, isSoldOut } } = this.props;
    return isLowQuantity && !isSoldOut;
  }

  renderProductMedia() {
    const { classes, product: { description } } = this.props;
    const chipClasses = { root: classes.chip, label: classes.chipLabel };
    const productStatus = this.productStatus;
    console.log("product status is", this.productStatus);
    return (
      <div className={classes.productMedia}>
        {productStatus && <Chip label={productStatus.label} classes={chipClasses} className={productStatus.style} />}
        {this.productLowQuantity && <Chip label={"Low Inventory"} classes={chipClasses} className={classes.warning} />}
        <img className={classes.img} src="http://via.placeholder.com/200" alt={description} />
      </div>
    );
  }

  renderProductInfo() {
    const { classes, product: { price, title, vendor } } = this.props;
    const { range: priceRange } = price || {};
    return (
      <div className={classes.productInfo}>
        <div>
          <Typography variant="body2">{title}</Typography>
          <Typography variant="body1">{vendor}</Typography>
        </div>

        <div>
          <Typography variant="body2">${priceRange || price}</Typography>
        </div>
      </div>
    );
  }

  render() {
    return (
      <div>
        {this.renderProductMedia()}
        {this.renderProductInfo()}
      </div>
    );
  }
}

export default ProductItem;
