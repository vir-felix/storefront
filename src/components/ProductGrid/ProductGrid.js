import React, { Component } from "react";
import PropTypes from "prop-types";
import Grid from "material-ui/Grid";

import ProductItem from "components/ProductItem";

class ProductGrid extends Component {
  static propTypes = {
    catalogItems: PropTypes.arrayOf(PropTypes.object)
  };

  renderProduct(edge) {
    const { node: { product, positions } } = edge;
    const weight = (positions.length) ? positions[0].displayWeight : 0;
    const { _id } = product;
    const gridItemSize = {
      0: {
        xs: 12,
        sm: 4,
        md: 3
      },
      1: {
        xs: 12,
        sm: 8,
        md: 6
      },
      2: {
        xs: 12,
        sm: 12,
        md: 9
      }
    };
    const gridItemProps = {
      key: _id,
      ...gridItemSize[weight]
    };
    return (
      <Grid item {...gridItemProps}>
        <ProductItem product={product} />
      </Grid>
    );
  }

  render() {
    const { catalogItems } = this.props;

    return (
      <section>
        <Grid container spacing={24}>
          {(catalogItems && catalogItems.length) ? catalogItems.map(this.renderProduct) : null }
        </Grid>
      </section>
    );
  }
}

export default ProductGrid;
