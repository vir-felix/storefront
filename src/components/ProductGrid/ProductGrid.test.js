import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "mobx-react";
import { MuiThemeProvider } from "material-ui/styles";
import theme from "lib/theme/reactionTheme";
import ProductGrid from "./ProductGrid";
import products from "./__mocks__/products.mock";

const uiStore = {
  appConfig: {
    publicRuntimeConfig: {
      externalAssetsUrl: "http://localhost:3000"
    }
  }
};

test("basic snapshot", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider uiStore={uiStore}>
        <ProductGrid catalogItems={products} primaryShopId="123" />
      </Provider>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
