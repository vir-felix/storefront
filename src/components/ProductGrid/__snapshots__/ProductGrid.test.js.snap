// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty product grid message 1`] = `
Array [
  <div
    className="MuiGrid-container-2 MuiGrid-spacing-xs-8-24 WithTracking-ProductGrid--filters-1"
  >
    <div
      className="MuiGrid-item-3"
    >
      <div
        className="MuiInputBase-root-124 MuiInput-root-111 Select-input-103"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-104"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-105 MuiSelect-selectMenu-108 Select-selectMenu-101 MuiInputBase-input-134 MuiInput-input-119"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            20 Products
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="page-size"
            name="pageSize"
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value={20}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-141 MuiSelect-icon-110"
            color={undefined}
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-3"
    >
      <div
        className="MuiInputBase-root-124 MuiInput-root-111 Select-input-103"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-104"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-105 MuiSelect-selectMenu-108 Select-selectMenu-101 MuiInputBase-input-134 MuiInput-input-119"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Newest
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="sort-by"
            name="sortBy"
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value="updatedAt-desc"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-141 MuiSelect-icon-110"
            color={undefined}
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="inject-ProductGridEmptyMessage-with-routingStore-root-155"
  >
    <p
      className="MuiTypography-root-158 MuiTypography-body1-167 MuiTypography-paragraph-177 inject-ProductGridEmptyMessage-with-routingStore-notFoundMessage-157"
    >
      Sorry! We couldn't find what you're looking for.
    </p>
    <p
      className="MuiTypography-root-158 MuiTypography-body1-167 inject-ProductGridEmptyMessage-with-routingStore-actionMessage-156"
    >
      <a
        className=""
        href="tag/test-tag"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        Clear filters
      </a>
    </p>
  </div>,
]
`;

exports[`basic snapshot 1`] = `
Array [
  <div
    className="MuiGrid-container-2 MuiGrid-spacing-xs-8-24 WithTracking-ProductGrid--filters-1"
  >
    <div
      className="MuiGrid-item-3"
    >
      <div
        className="MuiInputBase-root-124 MuiInput-root-111 Select-input-103"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-104"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-105 MuiSelect-selectMenu-108 Select-selectMenu-101 MuiInputBase-input-134 MuiInput-input-119"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            20 Products
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="page-size"
            name="pageSize"
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value={20}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-141 MuiSelect-icon-110"
            color={undefined}
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-3"
    >
      <div
        className="MuiInputBase-root-124 MuiInput-root-111 Select-input-103"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-104"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-105 MuiSelect-selectMenu-108 Select-selectMenu-101 MuiInputBase-input-134 MuiInput-input-119"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Newest
          </div>
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id="sort-by"
            name="sortBy"
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            rows={undefined}
            type="hidden"
            value="updatedAt-desc"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-141 MuiSelect-icon-110"
            color={undefined}
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-container-2 MuiGrid-spacing-xs-24-26"
  >
    <div
      className="CatalogGrid__GridContainer-iLBjFm bXBNVP"
    >
      <div
        className="CatalogGrid__GridItem-eahVxr FBXTD"
      >
        <CatalogGridItem
          currencyCode="USD"
          onClick={[Function]}
          placeholderImageURL="/static/placeholder.gif"
          product={
            Object {
              "_id": "cmVhY3Rpb24vY2F0YWxvZ1Byb2R1Y3Q6am5hOTVrYTlhTmtNeXRteGg=",
              "description": "Great jacket!",
              "isBackorder": false,
              "isLowQuantity": false,
              "isSoldOut": true,
              "pricing": Array [
                Object {
                  "compareAtPrice": null,
                  "currency": Object {
                    "code": "USD",
                  },
                  "displayPrice": "$12.99 - $19.99",
                  "price": 19.99,
                },
              ],
              "primaryImage": Object {
                "URLs": Object {
                  "large": "/assets/files/Media/aHEWna7yeBvNFM9bL/large/25580_CICN.jpg",
                  "medium": "/assets/files/Media/aHEWna7yeBvNFM9bL/medium/25580_CICN.jpg",
                  "small": "/assets/files/Media/aHEWna7yeBvNFM9bL/small/25580_CICN.png",
                  "thumbnail": "/assets/files/Media/aHEWna7yeBvNFM9bL/thumbnail/25580_CICN.png",
                },
                "priority": 0,
                "productId": "cmVhY3Rpb24vcHJvZHVjdDpqbmE5NWthOWFOa015dG14aA==",
                "toGrid": 1,
              },
              "slug": "men's-lightweigth-synchilla",
              "title": "Men's Lightweigth Synchilla",
              "vendor": "Patagonia",
            }
          }
        />
      </div>
      <div
        className="CatalogGrid__GridItem-eahVxr FBXTD"
      >
        <CatalogGridItem
          currencyCode="USD"
          onClick={[Function]}
          placeholderImageURL="/static/placeholder.gif"
          product={
            Object {
              "_id": "cmVhY3Rpb24vY2F0YWxvZ1Byb2R1Y3Q6QkNUTVo2SFR4RlNwcEpFU2s=",
              "description": "Sign in as administrator to edit.
You can clone this product from the product grid.
You can upload images click or drag in image box on the left here.
Tag this product below, and then add tag in navigation.
Click the bookmark in the tag to set product url.
Option variants, price, quantity, and child variants are created by clicking on the variant below, clone the variant to add more options.
Details can be added below the image for more specific product information.
 Login next to the cart, and then click the dashboard icon for more tools.",
              "isBackorder": false,
              "isLowQuantity": false,
              "isSoldOut": false,
              "pricing": Array [
                Object {
                  "compareAtPrice": null,
                  "currency": Object {
                    "code": "USD",
                  },
                  "displayPrice": "$12.99 - $19.99",
                  "price": 19.99,
                },
              ],
              "primaryImage": Object {
                "URLs": Object {
                  "large": "/assets/files/Media/mmY4m9gwPK2uFEZ8w/large/product_image.jpg",
                  "medium": "/assets/files/Media/mmY4m9gwPK2uFEZ8w/medium/product_image.jpg",
                  "small": "/assets/files/Media/mmY4m9gwPK2uFEZ8w/small/product_image.png",
                  "thumbnail": "/assets/files/Media/mmY4m9gwPK2uFEZ8w/thumbnail/product_image.png",
                },
                "priority": 0,
                "productId": "cmVhY3Rpb24vcHJvZHVjdDpCQ1RNWjZIVHhGU3BwSkVTaw==",
                "toGrid": 1,
              },
              "slug": "example-product",
              "title": "Basic Reaction Product",
              "vendor": "Reaction Commerce",
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="MuiGrid-container-2 MuiGrid-justify-xs-space-between-21 PageStepper-root-154"
  >
    <div
      className="MuiGrid-item-3"
    >
      <div
        className="Button__ButtonDiv-bnfrK cnBjlR"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title={undefined}
      >
        <div>
          Previous
        </div>
        <div
          className="Button__SpinnerWrap-bgOcuM cCKgPo"
          style={
            Object {
              "opacity": 0,
              "paddingLeft": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiGrid-item-3"
    />
  </div>,
]
`;
